index.xml

```diff
@@ -14044,66 +14181,126 @@ C implements A;</x:codeblock>
-                  </ol>
-                  And then, whether or not an exception was thrown:
-                  <ol>
-                    <li>Pop <var>script</var> off the <a class='dfnref external' href='http://www.whatwg.org/specs/web-apps/current-work/multipage/webappapis.html'>stack of incumbent scripts</a>.</li>
-                    <li>If an exception was thrown, end these steps, and allow it to propagate.</li>
-                  </ol>
-                </li>
-                </ol>
-                </li>
-                <li>If the callback function’s return type is <span class='idltype'>void</span>, return.</li>
-                <li>
-                  Return the result of <a class='dfnref' href='#dfn-convert-ecmascript-to-idl-value'>converting</a>
-                  <var>R</var> to an IDL value of the same type as the callback function’s return type.
-                </li>
+                <li>If the callback function's return type is <span class='idltype'>void</span>,
+                return.</li>
+
+                <li>Return the result of <a class='dfnref'
+                href='#dfn-convert-ecmascript-to-idl-value'>converting</a> <span
+                class='esvalue'>undefined</span> to the callback function's return type.</li>
               </ol>
-              And then, if an exception was thrown:
+            </li>
+
+            <li>Let <var>realm</var> be <var>F</var>'s <a class='dfnref'
+            href='#dfn-associated-realm'>associated Realm</a>.</li>
+
+            <li>Let <var>settings</var> be <var>realm</var>'s <a class='dfnref external'
+            href='https://html.spec.whatwg.org/multipage/webappapis.html#concept-realm-settings-object'>settings
+            object</a>.</li>
+
+            <li><a>Prepare to run script</a> with <var>settings</var>.</li>
+
+            <li>If <var>value</var>'s interface is not a <a class='dfnref'
+            href='#dfn-single-operation-callback-interface'>single operation callback interface</a>,
+            or if <a>!</a> <a>IsCallable</a>(<var>O</var>) is <span class='esvalue'>false</span>,
+            set <var>thisArg</var> to <var>O</var> (overriding the provided value).</li>
+
+            <li>Let <var>esArgs</var> be an empty List of ECMAScript values.</li>
+
+            <li>Let <var>i</var> be 0.</li>
+
+            <li>Let <var>count</var> be 0.</li>
+
+            <li>
+              While <var>i</var> &lt; <var>n</var>:
+
               <ol>
-                <li>If the callback function has a <a class='dfnref' href='#dfn-return-type'>return type</a> that is a <a href='#idl-promise'>promise type</a>, then:
+                <li>If <var>arg</var><sub><var>i</var></sub> is the special value “missing”, then
```

Comment:
Ah.  "missing" is what you get as the IDL value when `undefined` is passed to an optional argument with no default value.  It's handled in ES-to-IDL conversion for method calls, and for calls into a callback function; just not for the user object operation case.